lib/puppet/type/elasticsearch_component_template.rb

```diff
@@ -48,7 +48,7 @@ def insync?(value)
               val['template']['settings']['index'] = {} unless val['template']['settings'].key? 'index'
               (val['template']['settings'].keys - ['index']).each do |setting|
                 new_key = if setting.start_with? 'index.'
-                            setting[6..]
+                            setting[6..setting.length - 1]
```

Comment:
This is a longer way to say:
```ruby
setting[6..-1]
```

which is a longer way to say (original code):
```ruby
setting[6..]
```

What was intended here?

Reply:
- [6..-1] generate Rubo Cop:: Cop:: Style:: Slicing With Range error
- [6..] generate error issue https://github.com/voxpupuli/puppet-elasticsearch/issues/1222

I try to find a solution

Reply:
I see. This was caused by EOL Ruby bundled with EOL Puppet, better stick to "modern standards" rather than trying to be compatible with EOL products.  Older versions can still be used if necessary.